pkg/operator/controller/ingress/controller.go

```diff
@@ -1205,3 +1209,30 @@ func (r *reconciler) allRouterPodsDeleted(ingress *operatorv1.IngressController)
 
 	return true, nil
 }
+
+// updateInfraIngressLoadBalancerIPs updates GCP's PlatformStatus with Ingress LB IPs when the DNSType is `ClusterHosted`.
+func (r *reconciler) updateInfraIngressLoadBalancerIPs(service *corev1.Service, platformStatus *configv1.PlatformStatus) error {
+	if platformStatus != nil && platformStatus.Type == configv1.GCPPlatformType {
+		switch platformStatus.GCP.CloudLoadBalancerConfig.DNSType {
+		case configv1.ClusterHostedDNSType:
+			// The cluster is running its own CoreDNS pod for DNS. Update Infrastructure
+			// CR with the Ingress LB IPs. These values are used to configure the in-cluster
+			// DNS to provide resolution for *.apps.
+			ingresses := service.Status.LoadBalancer.Ingress
+			ingressLBIPs := []configv1.IP{}
+			for _, ingress := range ingresses {
+				if len(ingress.IP) > 0 {
+					ingressLBIPs = append(ingressLBIPs, configv1.IP(ingress.IP))
+				}
+			}
+			if platformStatus.GCP.CloudLoadBalancerConfig.ClusterHosted == nil {
+				platformStatus.GCP.CloudLoadBalancerConfig.ClusterHosted = &configv1.CloudLoadBalancerIPs{}
+			}
+			platformStatus.GCP.CloudLoadBalancerConfig.ClusterHosted.IngressLoadBalancerIPs = ingressLBIPs
+			return nil
```

Comment:
Are you missing a call to `r.client.Status().Update(...)` here?  And I think you'll need to add a permission to https://github.com/openshift/cluster-ingress-operator/blob/master/manifests/00-cluster-role.yaml:
```yaml
- apiGroups:
  - config.openshift.io
  resources:
  - infrastructures/status
  verbs:
  - update
```

Reply:
Oh, also, I think it would be appropriate to check whether `platformStatus.GCP.CloudLoadBalancerConfig.ClusterHosted.IngressLoadBalancerIPs` is already equal to `ingressLBIPs` and avoid doing a spurious update, assuming you do mean to call `r.client.Status().Update(...)` here.  I think the order doesn't matter, in which case you should do a comparison after sorting the slice, for example something like this:
```go
			ipCmpOpts := []cmp.Option{
				cmpopts.EquateEmpty(),
				cmpopts.SortSlices(func(a, b configv1.IP) bool {
					return a < b
				}),
			}
			if !cmp.Equal(platformStatus.GCP.CloudLoadBalancerConfig.ClusterHosted.IngressLoadBalancerIPs, ingressLBIPs, ipCmpOpts...) {
				return r.Status().Update(...)
			}
```
(You'll need to import `"github.com/google/go-cmp/cmp"` and `"github.com/google/go-cmp/cmp/cmpopts"`.)